eyecite/test_factories.py

```diff
@@ -98,11 +98,20 @@ def id_citation(source_text=None, index=0, **kwargs):
     return IdCitation(IdToken(source_text, 0, 99), index, **kwargs)
 
 
-def nonopinion_citation(source_text=None, index=0, **kwargs):
-    """Convenience function for creating mock NonopinionCitation objects."""
-    return NonopinionCitation(
-        SectionToken(source_text, 0, 99), index, **kwargs
+def unknown_citation(source_text=None, index=0, **kwargs):
+    """Convenience function for creating mock UnknownCitation objects."""
+    return UnknownCitation(SectionToken(source_text, 0, 99), index, **kwargs)
+
+
+def nonopinion_citation(*args, **kwargs):
+    from warnings import warn
+
+    warn(
+        """nonopinion_citation() will be deprecated in eyecite 2.5.0.
+        Please use unknown_citation() instead.""",
+        DeprecationWarning,
     )
+    return unknown_citation(*args, **kwargs)
```

Comment:
I don't think we need deprecation warnings in tests, but otherwise, this PR looks ready to me.

Reply:
This isn't in our tests per se -- the test factories are part of our public API that other people can use to write _their_ tests. So if we're going to alias the class I think we should alias this factory function too.

Reply:
OK, yeah, that *does* make sense. Merging!